dist/origin-web-common-services.js

```diff
@@ -554,6 +554,70 @@ angular.module('openshiftCommonServices')
 });
 ;'use strict';
 
+angular.module("openshiftCommonServices").
+service("ApplicationsService", function($filter, $q, DataService) {
+
+  var getApplications = function(context) {
+    var hasDeploymentFilter = $filter('hasDeployment');
+    var hasDeploymentConfigFilter = $filter('hasDeploymentConfig');
+    var deploymentConfigs;
+    var deployments;
+    var replicationControllers;
+    var replicaSets;
+    var statefulSets;
+    var deferred = $q.defer();
+
+    var resolveApplications = function() {
+      // Don't resolve until we have all the types
+      if (deploymentConfigs && deployments && replicationControllers && replicaSets && statefulSets) {
+        var apiObjects = deploymentConfigs.concat(deployments)
+          .concat(replicationControllers)
+          .concat(replicaSets)
+          .concat(statefulSets);
+        deferred.resolve(_.sortBy(apiObjects, ['metadata.name', 'kind']));
+      }
+    };
+
+    // Load all the "application" types
+    DataService.list('deploymentconfigs', context).then(function(deploymentConfigData) {
+      deploymentConfigs = _.toArray(deploymentConfigData.by('metadata.name'));
+      resolveApplications();
+    });
+    DataService.list('replicationcontrollers', context).then(function(replicationControllerData) {
+      replicationControllers = _.reject(replicationControllerData.by('metadata.name'), hasDeploymentConfigFilter);
```

Comment:
Can you open a follow-on issue to see if we use a label filter to exclude these in the `DataService.list` call?

If you're not pruning old deployments, you can potentially have a lot of old replication controllers for previous deployments. It would be nice to exclude these using the API rather than filtering on the client.

Reply:
Added issue #179